src/generateTasks.js

```diff
@@ -24,22 +32,18 @@ module.exports = async function generateTasks(config, gitDir, stagedRelFiles) {
       stagedFiles
         // Only worry about children of the CWD unless the pattern explicitly
         // specifies that it concerns a parent directory.
-        .filter(file => isParentDirPattern || pathIsInside(file, cwd))
+        .filter(file => isParentDirPattern || isPathInside(cwd, file))
         // Make the paths relative to CWD for filtering
         .map(file => path.relative(cwd, file)),
       pattern,
       {
-        ...globOptions,
-        ignore
-      }
-    ).map(file => {
-      // if you set relative option, then the file path will be relative to your package.json
-      if (config.relative) {
-        return path.normalize(file)
+        matchBase: !pattern.includes('/'),
```

Comment:
I’d add a comment here explaining what it does for future us :)

Reply:
Done.